x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/state/ModelState.java

```diff
@@ -43,9 +43,13 @@ public static String extractJobId(String docId) {
      * and ended with hash and an integer.
      */
     private static String v54ExtractJobId(String docId) {
-        Matcher matcher = V_5_4_DOC_ID_REGEX.matcher(docId);
+        int potentialSuffixIndex = docId.lastIndexOf('-');
+        if (potentialSuffixIndex <= 0 || potentialSuffixIndex >= docId.length() - 1) {
```

Comment:
Well spotted. I think it was actually a bug before, because a valid job ID cannot be an empty string. So previously this method could have returned an empty string as a valid job ID. In practice this was extremely unlikely, because no document ID that we expect to exist in the ML state index would begin with a hyphen. The only way it could happen is if somebody indexed some dodgy documents into this index. You'd have to be superuser to do this so in practice it doesn't happen.